crates/nu-protocol/src/pipeline_data/mod.rs

```diff
@@ -118,7 +121,7 @@ impl PipelineData {
     /// PipelineData doesn't always have a Span, but we can try!
     pub fn span(&self) -> Option<Span> {
         match self {
-            PipelineData::ListStream(..) => None,
+            PipelineData::ListStream(stream, ..) => Some(stream.span()),
```

Comment:
Mhh we could push that over the edge if we add a span to `Empty` to make sure a command can always refer to its input source. (there are some errors that are otherwise a bit hairy)

Reply:
Definitely worth considering / keeping in mind for the future.